File: backend/internal/models/supplier.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Supplier struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	Name        string         `gorm:"not null;index" json:"name" binding:"required"`
	Code        string         `gorm:"uniqueIndex;not null" json:"code" binding:"required"`
	Email       string         `gorm:"not null" json:"email" binding:"required,email"`
	Phone       string         `json:"phone"`
	Address     string         `json:"address"`
	City        string         `json:"city"`
	Country     string         `json:"country"`
	PostalCode  string         `json:"postal_code"`
	ContactName string         `json:"contact_name"`
	Rating      float32        `gorm:"type:decimal(3,2);default:0" json:"rating"`
	Status      string         `gorm:"type:varchar(20);default:'active'" json:"status"` // active, inactive, suspended
	Notes       string         `gorm:"type:text" json:"notes"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

func (Supplier) TableName() string {
	return "suppliers"
}

File: frontend/src/types/supplier.ts
export interface Supplier {
  id?: number
  name: string
  code: string
  email: string
  phone?: string
  address?: string
  city?: string
  country?: string
  postal_code?: string
  contact_name?: string
  rating?: number
  status?: string
  notes?: string
  created_at?: string
  updated_at?: string
}

export interface PaginationMeta {
  page: number
  page_size: number
  total: number
  total_pages: number
}

export interface SupplierListResponse {
  data: Supplier[]
  pagination: PaginationMeta
}

export interface SupplierResponse {
  data: Supplier
  message?: string
}
